File: app/src/main/java/ssilvalucas/testeandroidv2/screen/home/HomeConfigurator.java
package ssilvalucas.testeandroidv2.screen.home;

import java.lang.ref.WeakReference;

public enum HomeConfigurator {
    INSTANCE;

    public void configure(HomeActivity homeActivity) {

        HomeRouter router = new HomeRouter();
        router.activity = new WeakReference<>(homeActivity);

        HomePresenter presenter = new HomePresenter();
        presenter.output = new WeakReference<>(homeActivity);

        HomeInteractor interactor = new HomeInteractor();
        interactor.output = presenter;

        if(homeActivity.output == null){
            homeActivity.output = interactor;
        }
        if(homeActivity.router == null){
            homeActivity.router = router;
        }
    }

}

File: app/src/main/java/com/br/natanfelipe/bankapp/configurator/HomeConfigurator.kt
package com.br.natanfelipe.bankapp.configurator

import com.br.natanfelipe.bankapp.interactor.HomeInteractor
import com.br.natanfelipe.bankapp.presenter.HomePresenter
import com.br.natanfelipe.bankapp.view.home.HomeActivity
import java.lang.ref.WeakReference

enum class HomeConfigurator {
    INSTANCE;

    fun configure(activity: HomeActivity) {

        /*val router = LoginRouter()
        router.activity = WeakReference<HomeActivity>(activity)*/

        val presenter = HomePresenter()
        presenter.output = WeakReference(activity)

        val interactor = HomeInteractor()
        interactor.output = presenter

        if (activity.output == null) {
            activity.output = interactor
        }
        /*if (activity.router == null) {
            activity.router = router
        }*/
    }
}
